packages/lisk-p2p/src/p2p.ts

```diff
@@ -340,17 +337,20 @@ export class P2P extends EventEmitter {
 
 				const wsPort: number = parseInt(queryObject.wsPort, BASE_10_RADIX);
 				const peerId = constructPeerId(socket.remoteAddress, wsPort);
+				const queryOptions =
+					typeof queryObject.options === 'string'
+						? JSON.parse(queryObject.options)
+						: undefined;
 
-				const incomingPeerInfo: P2PDiscoveredPeerInfo = {
+				const incomingPeerInfo: P2PPeerInfo = {
 					ipAddress: socket.remoteAddress,
 					wsPort,
 					height: queryObject.height ? +queryObject.height : 0,
 					isDiscoveredPeer: true,
-					version: queryObject.version,
-					options:
-						typeof queryObject.options === 'string'
-							? JSON.parse(queryObject.options)
-							: undefined,
+					discoveredInfo: {
+						version: queryObject.version,
```

Comment:
prefer to put it in the root level, and make it mandatory.
maybe separate the type for `seed` and others?
tbd